src/peft/tuners/lora/layer.py

```diff
@@ -60,7 +65,8 @@ def _init_empty_weights(self, cls, *args, **kwargs) -> None:
         cls.__init__(self, *args, device="meta", **kwargs)
         self.to_empty(device=final_device)
 
-    def update_layer(self, adapter_name, r, lora_alpha, lora_dropout, init_lora_weights):
+    def update_layer(self, adapter_name, r, lora_alpha, lora_dropout,
+                     init_lora_weights: Optional[Union[str, bool]], weight_copy: Optional[torch.Tensor] = None):
```

Comment:
I think we can avoid the need to have `weight_copy` as a new argument to `update_layer` and all of the other changes related to that. This requires #1106 to be merged first. To summarize the change from that PR, it will result in the LoRA layers having a new attribute, `self.base_layer`, which is the reference to the original layer that we're adopting. You can already see that on the bnb layers. Therefore, inside of the `loftq_init_parameters` method, we will be able to fetch the weight by calling `self.base_layer.weight` (or, to be precise, `self.get_base_layer().weight`). I hope that the PR will be merged soon, it should make this PR easier overall.